Sources/web3swift/Utils/ENS/ENSResolver.swift

```diff
@@ -62,14 +62,29 @@ public extension ENS {
         }
 
         public func supportsInterface(interfaceID: Data) async throws -> Bool {
-            guard let supports = try? await supportsInterface(interfaceID: interfaceID.toHexString()) else {throw Web3Error.processingError(desc: "Can't get answer")}
+            guard let supports = try? await supportsInterface(interfaceID: interfaceID.toHexString()) else {
+                throw Web3Error.processingError(desc: "Can't get answer")
+            }
+            return supports
+        }
+
+        public func supportsInterface(interfaceID: InterfaceName) async throws -> Bool {
+            guard let supports = try? await supportsInterface(interfaceID: interfaceID.hash) else {
+                throw Web3Error.processingError(desc: "Can't get answer")
+            }
```

Comment:
Sorry, this one will goes out of control this way. I insist to clearly separate PRs with refactoring (cleanup) intention from any either feature or API related or bug fixing PRs. Like this bit of change it definitely not about code quality enhancement, this one provides new API as it seems. New API should be discussed at its own not with a bunch of refactoring.

E.g. this one change, and I think `writeOperation` enhancement should not be the part of this PR. Actually I'm not even sure that they should share single PR for both of them, maybe it's a good way to treat them separate each one. 

Just let me highlight it better, I'm not against this changes, I'm against including them within PR focused on refactoring stuff.

Reply:
I agree that gets way out of hand and very hard to review now but I guess with this one we can mostly pull through.

What if @pharms-eth creates a PR on top of this one for each separate feature/change? We review the individual PR and merge it in here? @yaroslavyaroslav

Reply:
@janndriessen This was my suggestion actually. Like to cut off both methods enhancement from this PR, e.g. leave that piece of code untouched, and pub that two changes in separate PR. So we could be sure that this one is just about improving bracelets and indenting, `self` word reducing, but nothing more.

I see that this is not a deal to hit the brake near to the end of this one, but I'd just caught myself, that with almost every new commit changes are increasing and I'm yet struggling to dig in them properly.

Reply:
The Code is combines now, so submitting different PRs at this stage is not trivial. It may be easier to redo everything at that point which is not in my interest

Reply:
It won't be too difficult to split. `git cherry-pick` can be used for that.

Reply:
Probably it was quite larger before https://github.com/web3swift-team/web3swift/pull/679/commits was merged as most of the commits are shared between #679 and this PR.